cnd/src/config/settings.rs

```diff
@@ -159,6 +163,115 @@ impl From<file::Logging> for Logging {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct Bitcoin {
+    pub network: ledger::Bitcoin,
+    pub bitcoind: Bitcoind,
+    pub fees: BitcoinFees,
+}
+
+impl Bitcoin {
+    fn default_from_network(network: ledger::Bitcoin) -> Self {
+        Self {
+            network,
+            bitcoind: Bitcoind::new(network),
+            fees: Default::default(),
+        }
+    }
+
+    pub fn from_file(
+        bitcoin: file::Bitcoin,
+        comit_network: Option<comit::Network>,
+    ) -> Result<Self> {
+        if let Some(comit_network) = comit_network {
+            let inferred = ledger::Bitcoin::from(comit_network);
+            if inferred != bitcoin.network {
+                anyhow::bail!(
+                    "inferred Bitcoin network {} from CLI argument {} but config file says {}",
+                    inferred,
+                    comit_network,
+                    bitcoin.network
+                );
+            }
+        }
+
+        let network = bitcoin.network;
+        let bitcoind = bitcoin.bitcoind.unwrap_or_else(|| Bitcoind::new(network));
+        let fees = bitcoin
+            .fees
+            .map_or_else(BitcoinFees::default, BitcoinFees::from);
+        Ok(Bitcoin {
+            network,
+            bitcoind,
+            fees,
+        })
+    }
+}
+
+impl From<Bitcoin> for file::Bitcoin {
+    fn from(settings: Bitcoin) -> Self {
+        Self {
+            network: settings.network,
+            bitcoind: Some(settings.bitcoind),
+            fees: Some(settings.fees.into()),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum BitcoinFees {
+    StaticSatPerVbyte(bitcoin::Amount),
+    CypherBlock(
+        /// Full URL to access blockcypher API blockchain endpoint: http://api.blockcypher.com/v1/btc/main
```

Comment:
This doc comment is getting a bit redundant. You could provide a newtype around `Url` and use that instead.

That would allow you to not repeat this documentation all over the place.

Reply:
Not convinced about the newtype, I'll clean up the comment.

Reply:
The idea of the newtype would be that the URL is statically linked to it and you for example can only instantiate it by passing a `comit::Network`. Up to you :)